File: Estruturas_pre-definidas/supermercados-java/Compra.java
import java.util.Scanner;
import java.util.ArrayList;

public class Compra {
    private ArrayList<Produto> produtos;
    private Usuario user;
    private ArrayList<Integer> quantidades;

    public Compra(Usuario user, ArrayList<Produto> prod, ArrayList<Integer> qtd){
        this.produtos = prod;
        this.quantidades = qtd;
        this.user = user;
    }

    public ArrayList<Produto> getProdutos() {
        return this.produtos;
    }

    public ArrayList<Integer> getQuantidades() {
        return this.quantidades;
    }   

    public Usuario getUsuario(){
        return this.user;
    }
}

File: Estruturas_pre-definidas/supermercados-cpp/compra.cpp
#include <string>
#include <vector>
#include "usuario.h"
#include "produto.h"
#include "compra.h"
using namespace std;


Compra::Compra(Usuario* user, vector<Produto*>& prod, vector<int>& qtd){
    this->user = user;
    this->produtos = prod;
    this->quantidades = qtd;
}

vector<Produto*>& Compra::getProdutos(){
    return this->produtos;
}

vector<int>& Compra::getQuantidades(){
    return this->quantidades;
}

Usuario* Compra::getUsuario(){
    return this->user;
}